### PR TITLE
Support new script/style behavior

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Implements [RFC0016](https://github.com/withastro/rfcs/blob/main/proposals/0016-style-script-defaults.md)
- Adds support for `is:inline` and `is:global`
- Companion PR to https://github.com/withastro/astro/pull/2961

## Testing

Tests updated

## Docs

Covered by core PR

### CODE REVIEW

File: internal/printer/print-to-js.go [hunk 1]
```diff
@@ -401,7 +401,7 @@ func render1(p *printer, n *Node, opts RenderOptions) {
 		p.print(`]`)
 	} else {
 		for _, a := range n.Attr {
-			if transform.IsImplictNodeMarker(a) {
+			if transform.IsImplictNodeMarker(a) || a.Key == "is:inline" {
```
Comment (natemoo-re) [inline internal/printer/print-to-js.go:404]: This strips `is:inline` directives from the output

File: internal/transform/scope-css.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ outer:
 		if n.DataAtom != a.Style {
 			continue
 		}
-		if hasTruthyAttr(n, "global") {
+		if hasTruthyAttr(n, "is:global") {
```
Comment (natemoo-re) [inline internal/transform/scope-css.go:22]: Removed support for `global`, just supporting `is:global`. Is that what we want? It's minimal to support both.

Comment (natemoo-re) [reply internal/transform/scope-css.go]: Added a migration warning for `global`, but still supporting both.

**Comment by FredKSchott** [comment]
LGTM!

**Pull request merged**